src/overlays/kaleido_scope/ovl_kaleido_scope/z_kaleido_debug.c

```diff
@@ -1,25 +1,1002 @@
+        KaleidoScope_DrawDigit(play, counterDigits[2], counterDigits[1], 212);
+
+        counterDigits[1] += 12;
+    }
+
+    // Oceanside Spider House Mask Order
+    for (counterDigits[1] = 217, i = 0; i < 6; i++) {
+        counterDigits[2] = gSaveContext.save.spiderHouseMaskOrder[i];
+        KaleidoScope_DrawDigit(play, counterDigits[2], counterDigits[1], 186);
+
+        counterDigits[1] += 12;
+    }
+
+    // Bombers code
+    for (counterDigits[1] = 220, i = 0; i < 5; i++) {
+        counterDigits[2] = gSaveContext.save.bomberCode[i];
+        KaleidoScope_DrawDigit(play, counterDigits[2], counterDigits[1], 210);
+
+        counterDigits[1] += 12;
+    }
+
+    // Draw a blue rectangle over the currently selected option
+    gDPPipeSync(POLY_OPA_DISP++);
+    gDPSetRenderMode(POLY_OPA_DISP++, G_RM_XLU_SURF, G_RM_XLU_SURF2);
+    gDPSetCombineMode(POLY_OPA_DISP++, G_CC_PRIMITIVE, G_CC_PRIMITIVE);
+    gDPSetPrimColor(POLY_OPA_DISP++, 0, 0, 0, 0, 200, 120);
+
+    gDPFillRectangle(POLY_OPA_DISP++, sSectionPositions[sCurSection].x, sSectionPositions[sCurSection].y,
+                     sSectionPositions[sCurSection].x + sSectionPositions[sCurSection].width,
+                     sSectionPositions[sCurSection].y + 15);
+
+    CLOSE_DISPS(play->state.gfxCtx);
+}
+
+void KaleidoScope_UpdateDebugEditor(PlayState* play) {
+    static s32 sPrevDBtnInput = 0;
+    static s32 sHeldDBtnTimer = 0;
+    PauseContext* pauseCtx = &play->pauseCtx;
+    Input* input = CONTROLLER1(&play->state);
+    s16 slot;
+    s16 value;
+    s32 dBtnInput = input->cur.button & (BTN_DUP | BTN_DDOWN | BTN_DLEFT | BTN_DRIGHT);
+
+    pauseCtx->stickRelX = input->rel.stick_x;
+    pauseCtx->stickRelY = input->rel.stick_y;
+
+    // Handles navigating the menu to different sections with the D-Pad
+    // When the same direction is held, registers the input periodically based on a timer
+    if (dBtnInput == sPrevDBtnInput) {
+        sHeldDBtnTimer--;
+        if (sHeldDBtnTimer < 0) {
+            sHeldDBtnTimer = 1;
+        } else {
+            dBtnInput ^= sPrevDBtnInput;
+        }
+    } else {
+        sPrevDBtnInput = dBtnInput;
+        sHeldDBtnTimer = 16;
+    }
+
+    if (CHECK_BTN_ANY(dBtnInput, BTN_DDOWN)) {
+        sCurRow++;
+        if (sCurRow > 17) {
+            sCurRow = 0;
+        }
+        sCurSection = sRowFirstSections[sCurRow];
+    } else if (CHECK_BTN_ANY(dBtnInput, BTN_DUP)) {
+        sCurRow--;
+        if (sCurRow < 0) {
+            sCurRow = 17;
+        }
+        sCurSection = sRowFirstSections[sCurRow];
+    } else if (CHECK_BTN_ANY(dBtnInput, BTN_DLEFT)) {
+        sCurSection--;
+        if (sCurSection < 0) {
+            sCurSection = 92;
+        }
+    } else if (CHECK_BTN_ANY(dBtnInput, BTN_DRIGHT)) {
+        sCurSection++;
+        if (sCurSection > 92) {
+            sCurSection = 0;
+        }
+    }
+
+    // Handles the logic to change values based on the selected section
+    switch (sCurSection) {
```

Comment:
Ideally this gets an enum, although it's likely that some of the values are not easy to discern.